test/extensions/common/aws/credentials_provider_impl_test.cc

```diff
@@ -2006,6 +2008,129 @@ TEST_F(TaskRoleCredentialsProviderUsingLibcurlTest, TimestampCredentialExpiratio
 }
 // End unit test for deprecated option using Libcurl client.
 
+// Specific test case for EKS Pod Identity, as Pod Identity auth token is only loaded at credential
+// refresh time
+class ContainerEKSPodIdentityCredentialsProviderTest : public testing::Test {
+public:
+  ContainerEKSPodIdentityCredentialsProviderTest()
+      : api_(Api::createApiForTest(time_system_)), raw_metadata_fetcher_(new MockMetadataFetcher) {
+    // Tue Jan  2 03:04:05 UTC 2018
+    time_system_.setSystemTime(std::chrono::milliseconds(1514862245000));
+  }
+
+  void setupProvider() {
+    scoped_runtime_.mergeValues(
+        {{"envoy.reloadable_features.use_http_client_to_fetch_aws_credentials", "true"}});
+    ON_CALL(context_, clusterManager()).WillByDefault(ReturnRef(cluster_manager_));
+    provider_ = std::make_shared<ContainerRoleCredentialsProvider>(
+        *api_, context_,
+        [this](Http::RequestMessage& message) -> absl::optional<std::string> {
+          return this->fetch_metadata_.fetch(message);
+        },
+        [this](Upstream::ClusterManager&, absl::string_view) {
+          metadata_fetcher_.reset(raw_metadata_fetcher_);
+          return std::move(metadata_fetcher_);
+        },
+        "169.254.170.2:80/path/to/doc", "", "credentials_provider_cluster");
```

Comment:
Do you want to set this to `169.254.170.23` as per https://docs.aws.amazon.com/eks/latest/userguide/pod-identities.html#pod-id-considerations ?

Reply:
Good catch :)